### ISSUE TITLE
[error]HttpError: Not Found

### ISSUE THREAD

**Issue by swuecho** [actions/labeler#16]
2019-08-30T04:47:25.3784277Z Current runner version: '2.157.2'
2019-08-30T04:47:25.3803572Z Prepare workflow directory
2019-08-30T04:47:25.4369293Z Prepare all required actions
2019-08-30T04:47:25.4675219Z Download action repository 'actions/labeler@v2'
2019-08-30T04:47:27.8427078Z ##[group]Run actions/labeler@v2
2019-08-30T04:47:27.8498581Z with:
2019-08-30T04:47:27.8498775Z   repo-token: ***
2019-08-30T04:47:27.8498811Z   configuration-path: .github/labeler.yml
2019-08-30T04:47:27.8498846Z ##[endgroup]
2019-08-30T04:47:28.4480335Z ##[error]HttpError: Not Found
2019-08-30T04:47:28.4807300Z ##[error]Not Found
2019-08-30T04:47:28.4832078Z ##[error]Node run failed with exit code 1
2019-08-30T04:47:28.5015714Z Cleaning up orphan processes

**Comment by swuecho** [comment]
totally lost. Could anyone give me a hint? thanks

**Comment by mansona** [comment]
I have hit this same issue when testing out the workflow 🤔 I don't know how public this is but here is the PR that is failing: https://github.com/mansona/actions-intro/pull/1

it is an empty repo that is just set up to test the action

**Comment by nfadili** [comment]
Running into this as well. I've tried every combination of refs I can think of: 
- `uses: actions/labeler@v2` - 404
- `uses: actions/labeler@releases/v2` - 404
- `uses: actions/labeler@master` - Doesn't have a `lib` folder

**Comment by gobadiah** [comment]
any news on this issue ?

**Comment by rohan-buchner** [comment]
@gobadiah I just bumped my head on this... ensure you have both the following files
```
.github/workflows/labeler.yml   // with the action
.github/labeler.yml        //with your labels
```

**Comment by gobadiah** [comment]
Thanks @rohan-buchner I already have the two files (I renamed the workflow from `label.yaml` to `labeler.yaml` just in case). The content of the workflow is copy pasted from the `README.md`, the content of the other file is just :

```
api: api/*
```

It was on a private repo, so I also tried doing this on a public repo, I get the same error.

**Comment by frextrite** [comment]
I am facing the same issue
```
##[error]HttpError: Resource not accessible by integration
##[error]Resource not accessible by integration
##[error]Node run failed with exit code 1
```

**Comment by NobbZ** [comment]
> ```plain
> ##[error]HttpError: Resource not accessible by integration
> ```

Thats actually #12.

**Comment by gilbertsoft** [comment]
Same issue here, but I'm facing some diffs of PR's from local repo or fork repos:

- fork repo: label gets **NOT** added and error is thrown
- local repo: there are two label actions executed, on the first run the error is thrown, second run is green and the label is added

![grafik](https://user-images.githubusercontent.com/25326036/70322510-8c507e80-182a-11ea-8ee9-a033ac5bd4a8.png)

![grafik](https://user-images.githubusercontent.com/25326036/70322537-a38f6c00-182a-11ea-9d7e-6fe8b7d54933.png)

**Comment by GMNGeoffrey** [comment]
I'm experiencing the same issue: https://github.com/GMNGeoffrey/iree/pull/10/checks?check_run_id=381938064

Could someone from the team please respond??

```
2020-01-09T18:07:48.5514225Z ##[section]Starting: Request a runner to run this job
2020-01-09T18:07:48.7204774Z Requesting a hosted runner in current repository's account/organization with labels: 'ubuntu-18.04', require runner match: True
2020-01-09T18:07:48.7854193Z Labels matched hosted runners has been found, waiting for one of them get assigned for this job.
2020-01-09T18:07:48.8029606Z ##[section]Finishing: Request a runner to run this job
2020-01-09T18:07:55.4244546Z Current runner version: '2.163.1'
2020-01-09T18:07:55.4245499Z Prepare workflow directory
2020-01-09T18:07:55.4488152Z Prepare all required actions
2020-01-09T18:07:55.4510698Z Download action repository 'actions/labeler@v2'
2020-01-09T18:07:57.9922248Z ##[group]Run echo "$AUTHOR"
2020-01-09T18:07:57.9922628Z [36;1mecho "$AUTHOR"[0m
2020-01-09T18:07:57.9964363Z shell: /bin/bash -e {0}
2020-01-09T18:07:57.9964628Z env:
2020-01-09T18:07:57.9964746Z   AUTHOR: GMNGeoffrey
2020-01-09T18:07:57.9964844Z ##[endgroup]
2020-01-09T18:07:58.0075385Z GMNGeoffrey
2020-01-09T18:07:58.0205498Z ##[group]Run actions/labeler@v2
2020-01-09T18:07:58.0205648Z with:
2020-01-09T18:07:58.0205823Z   repo-token: ***
2020-01-09T18:07:58.0205927Z   configuration-path: .github/labeler.yml
2020-01-09T18:07:58.0206032Z env:
2020-01-09T18:07:58.0206138Z   AUTHOR: GMNGeoffrey
2020-01-09T18:07:58.0206241Z ##[endgroup]
2020-01-09T18:07:58.8602042Z ##[error]HttpError: Not Found
2020-01-09T18:07:58.8604796Z ##[error]Not Found
2020-01-09T18:07:58.8610911Z ##[error]Node run failed with exit code 1
2020-01-09T18:07:58.8616908Z Cleaning up orphan processes
```

**Comment by GMNGeoffrey** [comment]
Well I'm an idiot. I forgot to `git add` the untracked .github/labeler.yml file before committing 🤦. I do think a more helpful error message here would be great

**Comment by dericlima** [comment]
The issue here is that the error message is not clear enough. The problem is the `.github/labeler.yml` is missing, so just add it inside the `.github` directory and you will be fine.

**Comment by rr-paras-patel** [comment]
Posting for awareness : This error message can be also be seen if you have `labeler.yaml ` instead of` labeler.yml`.

**Comment by smolanoteam** [comment]
does someone has a fix. Also, not working for me. Same error

**Comment by kingb** [comment]
Doh. I had created `.github/labeler.yaml` [WRONG] instead of `.github/labeler.yml` [Correct].

Suggestion:
1. Support .yaml as a valid extension
2. and/or make it explicit that no file was found at `.github/labeler.yml`

**Comment by Gornoka** [comment]
I have run into this error when copying workflows from one repo to another ( and forgetting the labeler.yml in the process). 

It would be great if the error message for a the missing config file was a little more explicit.

**Comment by sasankaweera123** [comment]
When I add the Create CODE_OF_CONDUCT.md then merge it. I'm getting this error also. Can anyone help 

Error: HttpError: Not Found
Error: Not Found

**Comment by Gornoka** [comment]
> When I add the Create CODE_OF_CONDUCT.md then merge it. I'm getting this error also. Can anyone help
> 
> Error: HttpError: Not Found Error: Not Found

I have seen this error when this action fails to find a file it requires, for me it was the config file (labeler.yaml) at the wrong location. I would make sure that you did not accidentaly moved / deleted / renamed that or any other file this action relies on.

**Comment by Shankar-CodeJunkie** [comment]
I too experience the same error, and I have created this preconfigured github action from the Github UI .   

```
Run actions/labeler@v4
  with:
    repo-token: ***
    configuration-path: .github/labeler.yml
    sync-labels: false
##[debug]fetching changed files for pr #2
##[debug]found changed files:
##[debug]  README.md
Error: HttpError: Not Found
Error: Not Found
```

**Comment by Gornoka** [comment]
> I too experience the same error, and I have created this preconfigured github action from the Github UI .
> 
> ```
> Run actions/labeler@v4
>   with:
>     repo-token: ***
>     configuration-path: .github/labeler.yml
>     sync-labels: false
> ##[debug]fetching changed files for pr #2
> ##[debug]found changed files:
> ##[debug]  README.md
> Error: HttpError: Not Found
> Error: Not Found
> ```

please make sure that the file `.github/labeler.yml` exists

**Comment by thiago-scherrer** [comment]
In the image below, I'm running a `cat` using the workspace environment variable. And the file exits:

![Sem título](https://user-images.githubusercontent.com/46422437/210101067-ca1065cd-9b7b-4424-97b4-c42b932ce76e.png)

**Comment by Gornoka** [comment]
> Below, I'm running a `cat` using the workspace environment variable. And the file exits:
> 
> ![Sem título](https://user-images.githubusercontent.com/46422437/210101067-ca1065cd-9b7b-4424-97b4-c42b932ce76e.png)

As far as I see it the action fetches the config directly from the github API through the client.rest.repos.getContent(...) function, I think the location should not use the '${GITHUB_WORKSPACE}'.
to validate that the not found error is related to the config file you could try to use [my fork branch](https://github.com/Gornoka/labeler/tree/error_message_for_missing_conf_rebase), which returns another error message in case that this specific file is not found.

**Comment by mac2000** [comment]
The last few messages and examples are indeed correct

Here is an short reproduction:

```yaml
- run: printf 'bug:\n- *\n' > /tmp/labeler.yml
- uses: actions/labeler@v4
  with:
    configuration-path: /tmp/labeler.yml
```

And because in code we have [this](https://github.com/actions/labeler/blob/main/src/labeler.ts#LL115C3-L122C23) it indeed will never ever work because even so it accepts the file path, it expects it to be committed

```ts
async function fetchContent(
  client: ClientType,
  repoPath: string
): Promise<string> {

  // add me to fix an issue
  if (fs.existsSync(repoPath)) {
    return Promise.resolve(fs.readFileSync(repoPath, 'utf8'));
  }

  const response: any = await client.rest.repos.getContent({
    owner: github.context.repo.owner,
    repo: github.context.repo.repo,
    path: repoPath,
    ref: github.context.sha
  });

  return Buffer.from(response.data.content, response.data.encoding).toString();
}
```

And it seems that there is already an pull request with similar fix - #394 so hopefully it will be merged somewhere in the future but for now it seems that there is no workarounds except of running some github script action and doing things by hands

**Comment by valeIT** [comment]
Same error here. The files seem to be named correctly though.

`.labeler.yml` in .github
```
merge when passing:
  - '*'
```

`.labeler.yml` in .github/workflows
```
name: "Pull Request Labeler"
on:
- pull_request_target

jobs:
  triage:
    permissions:
      contents: read
      pull-requests: write
    runs-on: ubuntu-latest
    steps:
    - uses: actions/labeler@v4
 ```

**Comment by ret708** [comment]
> Running into this as well. I've tried every combination of refs I can think of:
> 
> * `uses: actions/labeler@v2` - 404
> * `uses: actions/labeler@releases/v2` - 404
> * `uses: actions/labeler@master` - Doesn't have a `lib` folder

**Comment by ret708** [comment]
> The last few messages and examples are indeed correct
> 
> Here is an short reproduction:
> 
> ```yaml
> - run: printf 'bug:\n- *\n' > /tmp/labeler.yml
> - uses: actions/labeler@v4
>   with:
>     configuration-path: /tmp/labeler.yml
> ```
> 
> And because in code we have [this](https://github.com/actions/labeler/blob/main/src/labeler.ts#LL115C3-L122C23) it indeed will never ever work because even so it accepts the file path, it expects it to be committed
> 
> ```ts
> async function fetchContent(
>   client: ClientType,
>   repoPath: string
> ): Promise<string> {
> 
>   // add me to fix an issue
>   if (fs.existsSync(repoPath)) {
>     return Promise.resolve(fs.readFileSync(repoPath, 'utf8'));
>   }
> 
>   const response: any = await client.rest.repos.getContent({
>     owner: github.context.repo.owner,
>     repo: github.context.repo.repo,
>     path: repoPath,
>     ref: github.context.sha
>   });
> 
>   return Buffer.from(response.data.content, response.data.encoding).toString();
> }
> ```
> 
> And it seems that there is already an pull request with similar fix - #394 so hopefully it will be merged somewhere in the future but for now it seems that there is no workarounds except of running some github script action and doing things by hands
